chia/wallet/wallet.py

```diff
@@ -440,15 +455,16 @@ async def _generate_unsigned_transaction(
                     message_list.append(Coin(coin.name(), primary["puzzlehash"], primary["amount"]).name())
                 message: bytes32 = std_hash(b"".join(message_list))
                 puzzle: Program = await self.puzzle_for_puzzle_hash(coin.puzzle_hash)
+                assert puzzle.get_tree_hash() == coin.puzzle_hash
```

Comment:
I think this assertion is unnecessarily expensive, we should just trust that `puzzle_for_puzzle_hash` is doing its job.

Reply:
This is related to the puzzle decoration. We need to ensure the decorated puzzle is correct. Since the decorator is based on the config. This is more like a safety check for notice user's config is wrong.